### PR TITLE
CANVAS-143 - updgrade webpack to v5

### PR DESCRIPTION
**Pull request by jackvanb**
Upgrades webpack and webpack-cli to their latest versions.

For compatibility with instructure-ui, additional rules needed to be added since Webpack 5 supports `"type": "commonjs"/"type": "module"` in package.json. Instructure incorrectly includes esm files under a package.json with `type: "commonjs"`.

Additionally, additional environment variables need to be added for use in instructure-ui using EnvironmentPlugin. Webpack 5 removes support for using `process.env.MY_ENV_VAR` to access env variables.

### CODE REVIEW

File: .env.dist [hunk 1]
```diff
@@ -48,3 +48,6 @@ SECRET_VIDEORES_FTP_USERNAME=
 SECRET_VIDEORES_FTP_PWD=
 SECRET_VIDEORES_FILEPATH=
 VIDEORES_HASH_CLIENT_IP=true
+
+# InstructureUI
+DISABLE_SPEEDY_STYLESHEET=false
```
Comment (rlorenzo) [inline .env.dist:53]: @jackvanb This is always needed right? If so, then hardcode the value in webpack.config.js

Comment (rlorenzo) [reply .env.dist:53]: @jackvanb What does this setting do? I cannot find it documented anywhere

Comment (jackvanb) [reply .env.dist:53]: Looks like it is part of the glamor css (https://github.com/threepointone/glamor#speedy-mode). It's defined in Instructure ui-stylesheet (https://github.com/instructure/instructure-ui/blob/master/packages/ui-stylesheet/src/StyleSheet.js#L30).

Comment (rlorenzo) [reply .env.dist:53]: So DISABLE_SPEEDY_STYLESHEET needs to be in .env, because it reads from process.env.

Comment (rlorenzo) [reply .env.dist:53]: @jackvanb Is this setting needed?

I checked out your branch and ran yarn and then yarn dev without editing my .env file and the web app still seems to work.

**Comment by rlorenzo** [comment]
@jackvanb Have you tried this in production mode?

Here is what I did:

1. Edit .env and change mode to MODE=production
2. Set DEBUG=
3. Set DISABLE_SPEEDY_STYLESHEET=false

Then run yarn && yarn start

Load the app and I get an error Uncaught ReferenceError: process is not defined

In the bundle.js I see this are undefined:

- var Yn = process.env.ALWAYS_APPEND_UI_TESTABLE_LOCATORS ? L((function(e) 



- {

**Comment by jackvanb** [comment]
> @jackvanb Have you tried this in production mode?
> 
> Here is what I did:
> 
> 1. Edit .env and change mode to MODE=production
> 2. Set DEBUG=
> 3. Set DISABLE_SPEEDY_STYLESHEET=false
> 
> Then run yarn && yarn start
> 
> Load the app and I get an error Uncaught ReferenceError: process is not defined
> 
> In the bundle.js I see this are undefined:
> 
> * var Yn = process.env.ALWAYS_APPEND_UI_TESTABLE_LOCATORS ? L((function(e)

I'm not reproducing that error when going into production mode. It seems to be working fine for me.

**Comment by jackvanb** [comment]
@rlorenzo I was able to compile and run locally after just adding ALWAYS_APPEND_UI_TESTABLE_LOCATORS env variable on production mode.

**Comment by rlorenzo** [comment]
@jackvanb Thanks! This now works for me

**Pull request merged**